router/src/grpc/server.rs

```diff
@@ -264,6 +268,7 @@ impl TextEmbeddingsService {
         truncation_direction: tokenizers::TruncationDirection,
         raw_scores: bool,
         permit: OwnedSemaphorePermit,
+        batch_counter: Arc<AtomicUsize>,
```

Comment:
Looks like `batch_counter` here might be an optional type since `None` is passed to it at line 743!

```suggestion
        batch_counter: Optional<Arc<AtomicUsize>>,
```

Also, since `batch_counter` doesn't seem to be used here, maybe it'd make sense to rename it like this?

```suggestion
        _batch_counter: Optional<Arc<AtomicUsize>>,
```